src/DependencyInjection/Compiler/AddPackagesPass.php

```diff
@@ -20,58 +21,17 @@
  */
 class AddPackagesPass implements CompilerPassInterface
 {
-    /**
-     * @var string
-     */
-    private $jsonFile;
-
-    /**
-     * @param string $jsonFile
-     */
-    public function __construct(string $jsonFile)
-    {
-        $this->jsonFile = $jsonFile;
-    }
-
     /**
      * {@inheritdoc}
      */
     public function process(ContainerBuilder $container): void
     {
         $packages = [];
 
-        if (is_file($this->jsonFile)) {
-            $json = json_decode(file_get_contents($this->jsonFile), true);
-
-            if (null !== $json) {
-                $packages = $this->getVersions($json);
-            }
+        foreach (Versions::VERSIONS as $package => $version) {
+            $packages[$package] = substr($version, 0, strpos($version, '@'));
         }
 
         $container->setParameter('kernel.packages', $packages);
     }
-
-    /**
-     * Extracts the version numbers from the JSON data.
-     *
-     * @param array $json
-     *
-     * @return array
-     */
-    private function getVersions(array $json): array
-    {
-        $packages = [];
-
-        foreach ($json as $package) {
-            if (isset($package['version'])) {
-                if (isset($package['extra']['branch-alias'][$package['version']])) {
-                    $packages[$package['name']] = $package['extra']['branch-alias'][$package['version']];
```

Comment:
Does the `Versions` class handle branch aliases, too? Otherwise this is a BC break.

Reply:
It just uses what the `composer.lock` file contains. As far as I can tell, that does handle the branch aliases.